kong/db/declarative/init.lua

```diff
@@ -241,9 +241,8 @@ function Config:parse_table(dc_table, hash)
     error("expected a table as input", 2)
   end
 
-  on_the_fly_migration(dc_table)
-
   local entities, err_t, meta = self.schema:flatten(dc_table)
+  on_the_fly_migration(entities, dc_table._format_version)
```

Comment:
Could we move this line after `if err_t then ... end`?

Reply:
Yes, it should be after that.